src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -99,10 +102,35 @@ def _create_cssc_workflow(cmd, registry, schedule_cron_expression, resource_grou
         dry_run
     )
 
-    logger.warning(f"Deployment of {CONTINUOUS_PATCHING_WORKFLOW_NAME} tasks completed successfully.")
+    if not silent_execution:
+        print(f"Deployment of {CONTINUOUS_PATCHING_WORKFLOW_NAME} tasks completed successfully.")
+
+
+def _update_cssc_workflow(cmd, registry, schedule_cron_expression, resource_group, dry_run, task_list):
+    # compare the task definition to the existing tasks, if there is a difference, we need to update the tasks
+    # if we need to update the tasks, we will update the cron expression from it
+    # if not we just update the cron expression from the given parameter
+    for task in task_list:
+        deployed_task = task.step.encoded_task_content
+        extension_task = _create_encoded_task(CONTINUOUSPATCH_TASK_DEFINITION[task.name]["template_file"])
+        if deployed_task != extension_task:
+            logger.debug(f"Task {task.name} is different from the extension task, updating the task")
+
+            if schedule_cron_expression is None:
+                trigger_task = next((t for t in task_list if t.name == CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME), None)
+                if trigger_task is None:
+                    raise AzCLIError(f"Task {CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME} not found in the registry")
+                if not trigger_task.trigger.timer_triggers:
+                    raise AzCLIError(f"No timer triggers found for task {CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME}")
+                schedule_cron_expression = trigger_task.trigger.timer_triggers[0].schedule
 
+            _create_cssc_workflow(cmd, registry, schedule_cron_expression, resource_group, dry_run, silent_execution=True)
```

Comment:
Does _create_cssc_workflow update the current task or will redeploy all tasks in CONTINUOUSPATCH_TASK_DEFINITION?

Reply:
all tasks are redeployed, it is to keep the deployment operation simple (one template and one deployment request) at the expense of it asking to modify a task that does not require updating.

Reply:
Shall we update only the specific task instead of redeploying all tasks? This could avoid unnecessary ARM requests.

Reply:
changed logic to only update the task with diff, using task client instead of ARM deployment